### PR TITLE
QUA-948: Support .NET DotCover coverage reporting

### PR DESCRIPTION
**Pull request by camillof**
This PR adds support for uploading coverage reports generated by [DotCover](https://www.jetbrains.com/dotcover/). The report is expected to be located at the project root folder, and it must be named `dotcover.xml`

Please note that the `ReportType`  should be `DetailedXML` as it's the only format that prints coverage line by line.

```
    --ReportType=ARG              : (Optional) [HTML|JSON|XML|DetailedXML|NDependXML|SummaryXML].
                                    Specify to get a report of a certain type instead of a snapshot.
                                    For multiple reports, specify a list of report types separated by comma (,)
```

An example repo was uploaded here https://github.com/codeclimate/dot-net-coverage-test, and you can generate a test coverage report by running
```
dotnet build
dotnet dotcover test --dcReportType=DetailedXML --dcOutput="dotcover.xml" --no-build
```

### CODE REVIEW

File: cmd/format-coverage.go [hunk 1]
```diff
@@ -52,6 +53,7 @@ var formatterMap = map[string]formatters.Formatter{
 	"lcov-json":   &lcovjson.Formatter{},
 	"simplecov":   &simplecov.Formatter{},
 	"xccov":       &xccov.Formatter{},
+	"dotcover":       &dotcover.Formatter{},
```
Comment (larkinscott) [inline cmd/format-coverage.go:56]: This got formatted a little weird. Do we run `gofmt` against this repo?

**Review by larkinscott — APPROVED** [review]
This LGTM. Thanks for adding the integration tests. The only comment I might have about this would be: could this package and the `clover` package share any code? They seem relatively similar, though I wouldn't call it a blocker since we don't find ourselves adding formatters on a regular basis.

File: cmd/format-coverage.go (hunk 1 above)
Comment (camillof) [reply cmd/format-coverage.go]: Good catch, `codeclimate` is now set up on this repo 😉

**Comment by camillof** [comment]
> This LGTM. Thanks for adding the integration tests. The only comment I might have about this would be: could this package and the clover package share any code? They seem relatively similar, though I wouldn't call it a blocker since we don't find ourselves adding formatters on a regular basis.

Thanks for the review @larkinscott! Adding `codeclimate` to the repo allowed me to discover minor improvements. Regarding the similar looks among the formatters, I agree with that. Most of them share some code (for example, the `Search()` method is identical).

I'll try to refactor that, if possible, but I want to do it on a new PR, for clarity and to unblock this release for the customer asking for .NET support

**Pull request merged**